src/main/java/com/tabnine/binary/TabNineProcessFacade.java

```diff
@@ -1,39 +1,40 @@
 package com.tabnine.binary;
 
+import com.tabnine.DependencyContainer;
 import com.tabnine.exceptions.TabNineDeadException;
 
 import javax.annotation.Nonnull;
 import java.io.IOException;
 
 public class TabNineProcessFacade {
-    private static TabNineBinary tabNineBinary = new TabNineBinary();
+    private static BinaryFacade binaryFacade = DependencyContainer.instanceOfBinaryFacade();
 
-    public static void setTesting(TabNineBinary tabNineBinary) {
-        TabNineProcessFacade.tabNineBinary = tabNineBinary;
+    public static void setTesting(BinaryFacade tabNineBinary) {
+        TabNineProcessFacade.binaryFacade = tabNineBinary;
     }
 
-    public static void create() throws IOException {
-        tabNineBinary.create();
+    public static void create() throws IOException, NoValidBinaryToRunException {
```

Comment:
where the exceptions are handled?

Reply:
Here: https://github.com/codota/tabnine-intellij/blob/5d9f121b8557e28ac6147c8b738fc2406d7ec24d/src/main/java/com/tabnine/binary/TabNineGateway.java#L59